### ISSUE TITLE
Flashlist horizontal wrong height item

### ISSUE THREAD

**Issue by VictorPulzz** [Shopify/flash-list#881]
## Current behavior
Hi, first of all, you lib are cool, but I'm facing with issue, when I try use it in horizontal mode. Item of list are cutting in bottom
<img width="305" alt="Снимок экрана 2023-07-24 в 10 29 33" src="https://github.com/Shopify/flash-list/assets/23406402/279c0eb4-6a6c-4765-9da5-3839b56e7f5b">

https://github.com/Shopify/flash-list/assets/23406402/30ef17a5-28fd-4dab-90a7-9e2c574deb8d

code:
```ts
export const { width: WIDTH, height: HEIGHT } = Dimensions.get('window');

cont Comp = (props) => {
if({!data?.length) return <></>

return (
<FlashList
      ref={ref}
     data={data}
     renderItem=(({ item }) => <Card width={cardWidth} {...item} />)
      renderScrollComponent={ScrollView}
      bounces
     estimatedItemSize={WIDTH}
      showsVerticalScrollIndicator={false}
      showsHorizontalScrollIndicator={false}
      scrollEventThrottle={32}
      onEndReachedThreshold={2}
      {...props}
    />
)
}

```

## Expected behavior

Correct item size without cutting

## To Reproduce
Try to add different height for items

## Platform:

- [x ] iOS
- [ x] Android

## Environment
 "react-native": "0.71.8",
"@shopify/flash-list": "^1.4.2"

**Comment by MateuszRostkowski** [comment]
I'm also struggling with the same issue.

Here is a repro: https://snack.expo.dev/@mateuszrostkowski/flashlist-screen-orientation-switch-bug

I've got problem after changing orientation the item size is cut in the middle.

https://github.com/Shopify/flash-list/assets/38006370/767c1d11-9c58-4965-af28-d67de01ec82a

**Comment by QOAL** [comment]
I've been experiencing FlashList's height only ever updating when the new height is smaller.
I had a poke around, and onLayout will follow this trend of only firing when the new height is smaller.
So I dug a bit deeper in to the RecyclingListView that FlashList effectively wraps, and I think I've found a workaround for this, if you know the desired height:
```Javascript
useEffect(() => {
	YourFlashListRef.current.rlvRef?._onSizeChanged({
		width: YourFlashListRef.current.rlvRef._layout.width,
		height: desiredHeight
	})
}, [desiredHeight, YourFlashListRef])
```
I don't know if the issue lies with RecyclingListView rather than FlashList, but I hope this helps with any further investigation.

**Comment by ammar-madni** [comment]
> I've been experiencing FlashList's height only ever updating when the new height is smaller. I had a poke around, and onLayout will follow this trend of only firing when the new height is smaller. So I dug a bit deeper in to the RecyclingListView that FlashList effectively wraps, and I think I've found a workaround for this, if you know the desired height:
> 
> ```js
> useEffect(() => {
> 	YourFlashListRef.current.rlvRef?._onSizeChanged({
> 		width: YourFlashListRef.current.rlvRef._layout.width,
> 		height: desiredHeight
> 	})
> }, [desiredHeight, YourFlashListRef])
> ```
> 
> I don't know if the issue lies with RecyclingListView rather than FlashList, but I hope this helps with any further investigation.

Thanks for this, it's worked for me. My only issue is I don't want to use a fixed height for my list as my data usually determines the height. Any idea the best way to calculate what the correct height should be?

**Comment by VictorPulzz** [comment]
@ammar-madni You welcome:)

**Comment by dan-pugsley** [comment]
> For me the issue was that my `EmptyListComponent` height was smaller than my `renderItem` height. As soon as I set the `EmptyListComponent` to the same height as my items everything worked as expected.

Similar for me, I had an image which loaded a few frames after the first render of FlashList, changing the height. I just needed to set an explicit height for the image so the layout was correct on the first render.

**Comment by ammar-madni** [comment]
@dan-pugsley @davidcognite interesting!

I very much appreciate you putting your solutions here.

I'll be revisiting this issue to see if either of those resolve the issue as both of those sound relevant to my scenario.

**Comment by carlos-dubon** [comment]
This fixed the problem for me. I recommend first finding what is your avg. item height. The only downside is it shifts the layout on initial render...

```tsx
import {FlashList, FlashListProps} from '@shopify/flash-list';
import {useEffect, useRef, useState} from 'react';
import {View} from 'react-native';

interface HorizontalFlashListComponentProps<T> extends FlashListProps<T> {
  estimatedItemHeight: number;
}

export const HorizontalFlashListComponent = <T extends unknown>(
  props: HorizontalFlashListComponentProps<T>,
) => {
  const [maxItemHeight, setMaxItemHeight] = useState<number>(
    props.estimatedItemHeight,
  );

  const flashListRef = useRef<FlashList<T>>(null);

  // FIXME: See https://github.com/Shopify/flash-list/issues/881
  useEffect(() => {
    // @ts-ignore: Bypass private property access warning
    flashListRef.current?.rlvRef?._onSizeChanged({
      // @ts-ignore: Bypass private property access warning
      width: flashListRef.current.rlvRef._layout.width,
      height: maxItemHeight,
    });
  }, [maxItemHeight]);

  return (
    <FlashList
      {...props}
      ref={flashListRef}
      horizontal
      showsHorizontalScrollIndicator={false}
      renderItem={listItem => {
        return (
          <View
            onLayout={layoutEvent => {
              const itemHeight = layoutEvent.nativeEvent.layout.height;

              if (itemHeight > maxItemHeight) {
                setMaxItemHeight(itemHeight);
              }
            }}>
            {props.renderItem?.(listItem)}
          </View>
        );
      }}
    />
  );
};

```

**Comment by VictorPulzz** [comment]
@carlos-dubon this is all cool and interesting and looks great, but it would be good if this was fixed inside the plugin, otherwise it has been weighing for more than a year :)

**Comment by carlos-dubon** [comment]
@VictorPulzz yeah, I agree. Let's hope they don't close the issue at least. (Next.js likes to do that when community solutions are proposed)

**Comment by RaulFaife** [comment]
Having the same issue. I'll try the solutions I've seen here but I hope it's resolved

![Image](https://github.com/user-attachments/assets/73b06445-c6dc-45d3-9787-fe8164b9320d)

**Comment by RobinBobin** [comment]
Too sad, this isn't solved 🙁

**Comment by RobinBobin** [comment]
Displaying an image in my items makes items cut in height, as others mentioned.

**Comment by woltsu** [comment]
[This](https://github.com/Shopify/flash-list/issues/581#issuecomment-2122004124) fixed it for me!

**Comment by naqvitalha** [comment]
Folks, this should be resolved in v2. Can someone test their use case in new arch + v2 and let me know if it's resolved. We've made many improvements to horizontal lists.

**Comment by Francismb** [comment]
> Folks, this should be resolved in v2. Can someone test their use case in new arch + v2 and let me know if it's resolved. We've made many improvements to horizontal lists.

Our app is on V2 and seems to have the same issue or at least a similar one. To me it looks like the footer and header components render first and once the data loads, we display cards horizontally that may be taller than the footer and header components.
